### ISSUE TITLE
UP031 does not flag errors

### ISSUE THREAD

**Issue by abhijeetbodas2001** [astral-sh/ruff#8902]
The `UP031` fixer seems to be broken. I picked up one of the examples from the documentation for the fixer itself (https://docs.astral.sh/ruff/rules/printf-string-formatting/), but latest `ruff v0.1.6` did not flag it.

```bash
[~] $ cat test.py
[~] $ cat -n test.py
     1  value = 1
     2  print("%s" % value)  # "1"
     3  print("{}".format(value))  # "1"
     4
     5  value = (1,)
     6  print("%s" % value)  # "1"
     7  print("{}".format(value))  # "(1,)"
     8

[~] $ ruff --select UP test.py
test.py:3:7: UP032 [*] Use f-string instead of `format` call
test.py:7:7: UP032 [*] Use f-string instead of `format` call
Found 2 errors.
[*] 2 fixable with the `--fix` option.
[~] $ ruff --version
ruff 0.1.6
[~] $
```

(Notice that line 2 and line 6 did not get flagged by `UP031`)

Please LMK if any other info is required for debugging. Thanks a lot for all your efforts on this project!

**Comment by Avasam** [comment]
I think you meant `UP031`.  I don't think Ruff currently infers the type of the variable. The example in the doc shows you why Ruff can't autofix this. It's expected to not be raised with that specific example.  Although I think it should still raise it, see #6796, which this issue might duplicate.

**Comment by abhijeetbodas2001** [comment]
> I think you meant UP031

Oops, yes. I will update the title.

> Although I think it should still raise it, see https://github.com/astral-sh/ruff/issues/6796, which this issue might duplicate.

Yup, that issue describes the same use case. Even I wanted just flagging the occurence, even if not autofixing it.
I will close out this issue. Thanks!

**Comment by abhijeetbodas2001** [comment]
Duplicate of https://github.com/astral-sh/ruff/issues/6796.
